include/pinocchio/spatial/inertia.hpp

```diff
@@ -889,6 +960,366 @@ namespace pinocchio
+    {
+      LogCholeskyParameters params(log_cholesky);
+      Vector10 dynamic_params = params.toDynamicParameters();
+      return FromDynamicParameters(dynamic_params);
+    }
+
+    void disp_impl(std::ostream & os) const
+    {
+      os << "  matrix = \n"
+         << matrix << "\n"
+         << "  m = " << mass << "\n"
+         << "  h = " << h.transpose() << "\n"
+         << "  sigma = \n"
+         << Sigma << "";
+    }
+
+    friend std::ostream & operator<<(std::ostream & os, const PseudoInertiaTpl & pi)
+    {
+      pi.disp_impl(os);
+      return os;
+    }
+  };
+
+  /**
+   * @brief A structure representing log Cholesky parameters.
+   *
+   * References:
+   * - Rucker, Caleb, and Patrick M. Wensing. "Smooth parameterization of rigid-body inertia."
+   * IEEE Robotics and Automation Letters 7.2 (2022): 2771-2778.
+   */
+  template<typename Scalar, int Options>
+  struct LogCholeskyParametersTpl
+  {
+    typedef Eigen::Matrix<Scalar, 10, 1, Options> Vector10;
+    typedef Eigen::Matrix<Scalar, 10, 10, Options> Matrix10;
+    typedef PseudoInertiaTpl<Scalar, Options> PseudoInertia;
+
+    Vector10 parameters; ///< 10-dimensional vector of log Cholesky parameters
+
+    Scalar alpha;
+    Scalar d1;
+    Scalar d2;
+    Scalar d3;
+    Scalar s12;
+    Scalar s23;
+    Scalar s13;
+    Scalar t1;
+    Scalar t2;
+    Scalar t3;
+    /**
+     * @brief Constructor for LogCholeskyParametersTpl.
+     * @param log_cholesky A 10-dimensional vector of log Cholesky parameters.
+     */
+    explicit LogCholeskyParametersTpl(const Vector10 & log_cholesky)
+    : parameters(log_cholesky)
+    , alpha(log_cholesky[0])
+    , d1(log_cholesky[1])
+    , d2(log_cholesky[2])
+    , d3(log_cholesky[3])
+    , s12(log_cholesky[4])
+    , s23(log_cholesky[5])
+    , s13(log_cholesky[6])
+    , t1(log_cholesky[7])
+    , t2(log_cholesky[8])
+    , t3(log_cholesky[9])
+    {
+    }
+    /**
+     * @brief Converts the LogCholeskyParametersTpl object to dynamic parameters.
+     * @return A 10-dimensional vector of dynamic parameters.
+     */
+    Vector10 toDynamicParameters() const
+    {
+      Vector10 dynamic_params;
+
+      const Scalar exp_d1 = math::exp(d1);
+      const Scalar exp_d2 = math::exp(d2);
+      const Scalar exp_d3 = math::exp(d3);
+
+      dynamic_params[0] = 1;
+      dynamic_params[1] = t1;
+      dynamic_params[2] = t2;
+      dynamic_params[3] = t3;
+      dynamic_params[4] = s23 * s23 + t2 * t2 + t3 * t3 + exp_d2 * exp_d2 + exp_d3 * exp_d3;
+      dynamic_params[5] = -s12 * exp_d2 - s13 * s23 - t1 * t2;
+      dynamic_params[6] =
+        s12 * s12 + s13 * s13 + t1 * t1 + t3 * t3 + exp_d1 * exp_d1 + exp_d3 * exp_d3;
+      dynamic_params[7] = -s13 * exp_d3 - t1 * t3;
+      dynamic_params[8] = -s23 * exp_d3 - t2 * t3;
+      dynamic_params[9] =
+        s12 * s12 + s13 * s13 + s23 * s23 + t1 * t1 + t2 * t2 + exp_d1 * exp_d1 + exp_d2 * exp_d2;
```

Comment:
You can use math::pow improve the readability (you can also declare `using math::pow` at the beginning of this function to have a shorter function call).

You can also use the `// clang-format off` and `// clang-format on` comment to improve the formatting.